src/main/java/core/basesyntax/utils/transaction/handlers/impl/PurchaseOperation.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.utils.transaction.handlers.impl;
+
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.transaction.FruitTransaction;
+import core.basesyntax.utils.transaction.handlers.OperationHandler;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void perform(FruitTransaction transaction, StorageImpl storage) {
+        storage.updateEntry(transaction.getProduct(), -transaction.getQuantity());
+    }
```

Comment:
Handling Purchase operation.
Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative. Currently, there is no check to ensure that the quantity being purchased does not exceed the available quantity in storage, which can lead to a negative balance for the product.